chain/index/gc_test.go

```diff
@@ -11,57 +11,66 @@ import (
 
 func TestGC(t *testing.T) {
 	ctx := context.Background()
-	rng := pseudo.New(pseudo.NewSource(time.Now().UnixNano()))
+	genesisTime := time.Now()
+	rng := pseudo.New(pseudo.NewSource(genesisTime.UnixNano()))
 
 	// head at height 60
 	// insert tipsets at heigh 1,10,50.
 	// retention epochs is 20
+	// tipset at height 60 will be in the future
 	si, _, _ := setupWithHeadIndexed(t, 60, rng)
 	si.gcRetentionEpochs = 20
 	defer func() { _ = si.Close() }()
 
-	tsCid1 := randomCid(t, rng)
-	tsCid10 := randomCid(t, rng)
-	tsCid50 := randomCid(t, rng)
-
-	insertTipsetMessage(t, si, tipsetMessage{
-		tipsetKeyCid: tsCid1.Bytes(),
-		height:       1,
-		reverted:     false,
-		messageCid:   randomCid(t, rng).Bytes(),
-		messageIndex: 0,
-	})
-
-	insertTipsetMessage(t, si, tipsetMessage{
-		tipsetKeyCid: tsCid10.Bytes(),
-		height:       10,
-		reverted:     false,
-		messageCid:   randomCid(t, rng).Bytes(),
-		messageIndex: 0,
-	})
-
-	insertTipsetMessage(t, si, tipsetMessage{
-		tipsetKeyCid: tsCid50.Bytes(),
-		height:       50,
-		reverted:     false,
-		messageCid:   randomCid(t, rng).Bytes(),
-		messageIndex: 0,
-	})
+	// all tipsets will be in future
+	tsKeyCid1, _, _ := insertRandomTipsetAtHeight(t, si, 1, false, genesisTime)
+	tsKeyCid2, _, _ := insertRandomTipsetAtHeight(t, si, 10, false, genesisTime)
+	tsKeyCid3, _, _ := insertRandomTipsetAtHeight(t, si, 50, false, genesisTime)
 
 	si.gc(ctx)
 
-	// tipset at height 1 and 10 should be removed
+	// tipset at height 1 data should be removed
 	var count int
 	err := si.db.QueryRow("SELECT COUNT(*) FROM tipset_message WHERE height = 1").Scan(&count)
 	require.NoError(t, err)
 	require.Equal(t, 0, count)
 
+	err = si.stmts.getNonRevertedTipsetEventCountStmt.QueryRow(tsKeyCid1.Bytes()).Scan(&count)
+	require.NoError(t, err)
+	require.Equal(t, 0, count)
+
+	err = si.stmts.getNonRevertedTipsetEventEntriesCountStmt.QueryRow(tsKeyCid1.Bytes()).Scan(&count)
+	require.NoError(t, err)
+	require.Equal(t, 0, count)
+
+	// tipset at height 10 data should be removed
 	err = si.db.QueryRow("SELECT COUNT(*) FROM tipset_message WHERE height = 10").Scan(&count)
 	require.NoError(t, err)
 	require.Equal(t, 0, count)
 
+	err = si.stmts.getNonRevertedTipsetEventCountStmt.QueryRow(tsKeyCid2.Bytes()).Scan(&count)
+	require.NoError(t, err)
+	require.Equal(t, 0, count)
+
+	err = si.stmts.getNonRevertedTipsetEventEntriesCountStmt.QueryRow(tsKeyCid2.Bytes()).Scan(&count)
+	require.NoError(t, err)
+	require.Equal(t, 0, count)
+
 	// tipset at height 50 should not be removed
 	err = si.db.QueryRow("SELECT COUNT(*) FROM tipset_message WHERE height = 50").Scan(&count)
```

Comment:
lot of duplication here. Can we wrap this and the two sql statements below into a function and re-use it everywhere ?